docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,389 @@
+.. _custom_core:
+
+Adding a custom core
+====================
+
+You may want to add a custom RISC-V core to Chipyard generator. If the top module of your core is not in Chisel, 
+you will first need to create a Verilog blackbox for it. See :ref:`incorporating-verilog-blocks` for instructions.
+Once you have a top module in Chisel, you are ready to create integrate it with Chipyard. 
+
+``generators/ariane/src/main/scala/ArianeTile.scala`` and ``generators/boom/src/main/scala/common/tile.scala`` 
+provide two examples of how to integrate a core.
+
+.. note:: 
+
+    RoCC is not supported by custom core currently. Please use Rocket or Boom as the RoCC base core if you need to use RoCC.
+
+Parameter Case Classes
+----------------------
+
+Chipyard will generate a core for every ``InstantiableTileParams`` object it discovered in the current config.
+This object is derived from``TileParams``, a trait containing the information needed to create a tile. All cores must have
+their own implementation of ``InstantiableTileParams``, as well as ``CoreParams`` which is passed as a field in ``TileParams``.
+
+``TileParams`` holds the parameters for the tile, which are the same for every generated core, while ``CoreParams`` 
+contains the parameters for individual cores. They must be implemented as case classes with fields that can be overridden by 
+other config fragments as the constructor parameters. See the appendix at the bottom of the page for a list of 
```

Comment:
Can you put a markdown reference label and link there?

Reply:
Jerry suggest that I should remove the appendix and add it later (until we know what to list there), so I think I am just removing the last sentence.

Reply:
sure